builtin/providers/aws/resource_aws_s3_bucket_test.go

```diff
@@ -145,6 +172,38 @@ func testAccCheckAWSS3BucketExists(n string) resource.TestCheckFunc {
 	}
 }
 
+func testAccCheckAWSS3BucketPolicy(n string, policy string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, _ := s.RootModule().Resources[n]
+		conn := testAccProvider.Meta().(*AWSClient).s3conn
+
+		out, err := conn.GetBucketPolicy(&s3.GetBucketPolicyInput{
+			Bucket: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			if policy == "" {
+				// If we want to assert that the policy is not there, than
+				// this error is expected
+				return nil
+			} else {
+				return fmt.Errorf("GetBucketPolicy error: %v", err)
+			}
+		}
+
+		if v := out.Policy; v == nil {
+			if policy != "" {
+				return fmt.Errorf("bad policy, found nil, expected: %s", policy)
```

Comment:
This currently fails due to https://github.com/awslabs/aws-sdk-go/issues/236

I believe @m-s-austin also ran into this in https://github.com/iJoinSolutions/terraform/commit/cd45e45d6d04a3956fe35c178d5e816ba18d1051